dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavResult.java

```diff
@@ -0,0 +1,374 @@
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public String getTarget() {
+        if (type.equals("link"))
+            return target;
+        else
+            return "_self";
+    }
+
+    public void setTarget(String target) {
+
+        this.target = target;
+    }
+
+    public String getPermissionId() {
+        return permissionId;
+    }
+
+    public void setPermissionId(String permissionId) {
+        this.permissionId = permissionId;
+    }
+
+
+    /// Permissionable methods ///
+
+    @Override
+    public String getOwner() {
+        return sysuser.getUserId();
+    }
+
+    @Override
+    public void setOwner(String owner) {}
+
+    @Override
+    public List<PermissionSummary> acceptedPermissions() {
+        return Arrays.asList(new PermissionSummary[] {new PermissionSummary("READ", "READ", PermissionAPI.PERMISSION_READ)});
+    }
+
+    @Override
+    public List<RelatedPermissionableGroup> permissionDependencies(int requiredPermission) {
+        return null;
+    }
+
+    @Override
+    public Permissionable getParentPermissionable() throws DotDataException {
+        try {
+            if (type.equals("htmlpage")) {
+                Identifier ident = APILocator.getIdentifierAPI()
+                    .find(permissionId);
+                if (Identifier.ASSET_TYPE_CONTENTLET.equalsIgnoreCase(ident.getAssetType()))
+                    return APILocator.getContentletAPI()
+                        .findContentletByIdentifier(permissionId, true, APILocator.getLanguageAPI()
+                            .getDefaultLanguage()
+                            .getId(), sysuser, false)
+                        .getParentPermissionable();
+            }
+            if (type.equals("folder"))
+                return APILocator.getFolderAPI()
+                    .find(folderId, sysuser, false)
+                    .getParentPermissionable();
+            if (type.equals("link"))
+                return APILocator.getMenuLinkAPI()
+                    .findWorkingLinkById(permissionId, sysuser, false)
+                    .getParentPermissionable();
+            if (type.equals("file")) {
+                Identifier ident = APILocator.getIdentifierAPI()
+                    .find(permissionId);
+                if (Identifier.ASSET_TYPE_CONTENTLET.equalsIgnoreCase(ident.getAssetType()))
+                    return APILocator.getContentletAPI()
+                        .findContentletByIdentifier(permissionId, true, APILocator.getLanguageAPI()
+                            .getDefaultLanguage()
+                            .getId(), sysuser, false)
+                        .getParentPermissionable();
+            }
+            return null;
+        } catch (DotSecurityException e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)